pallets/msa/src/lib.rs

```diff
@@ -112,13 +130,48 @@ pub mod pallet {
 		pub fn create(origin: OriginFor<T>) -> DispatchResult {
 			let who = ensure_signed(origin)?;
 
-			let (identifier, _) = Self::create_account(who.clone())?;
+			let (identifier, _) = Self::create_account(who.clone(), EMPTY_FUNCTION)?;
 
 			Self::deposit_event(Event::MsaCreated { msa_id: identifier, key: who });
 
 			Ok(())
 		}
 
+		#[pallet::weight(T::WeightInfo::create(10_000))]
+		pub fn create_account_with_delegate(
+			origin: OriginFor<T>,
+			proof: MultiSignature,
```

Comment:
your missing the key of the signer?

Reply:
Was not sure if the old structure was making sense or not. I can modify it to be compatible with the rest

Reply:
Yeah the reason that I pulled out was to make it more practical to use. It also actually simplified the code.